python/cucim/src/cucim/core/operations/morphology/_pba_2d.py

```diff
@@ -1,8 +1,40 @@
+import functools
 import math
+import numbers
 import os
 
+
 import cupy
 
+try:
+    # math.lcm was introduced in Python 3.9
+    from math import lcm
+except ImportError:
+
+    """Fallback implementation of least common multiple (lcm)
+
+    TODO: remove once minimum Python requirement is >= 3.9
+    """
+
+    def _lcm(a, b):
+        return abs(b * (a // math.gcd(a, b)))
+
+    @functools.lru_cache()
+    def lcm(*args):
```

Comment:
Curious what the motivation is for caching here

Reply:
When using `math.lcm` directly in Python 3.8+ it is much faster (~150 ns vs. 2.8 µs for this fallback) so the cache was just to avoid that overhead prior to kernel launch. I think in practice it is not very important and can remove it if you prefer.

I figured often the (m1, m2, m3) block parameters would usually be the same. This function is not used at all when `block_params` is left at the default of None, though.